libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -454,6 +457,19 @@ class MapboxRouteLineApi(
             )
         }
 
+        ifNonNull(lastLocationPoint) {
+            val distance = TurfMeasurement.distance(it, point, TurfConstants.UNIT_METERS)
```

Comment:
Have you checked the performance difference? Because `distance` is not free, at least we had complaints about its performance (see NAVSDK-757).

Reply:
I'd imagine the benefits could outweigh the penalty but this is a good shout, let's measure the difference.

Reply:
Well, given how often we'd call this, it might not be worth it after all. Perhaps we could only check for location equality for starters. This should cover cases of stable GPS signal and being stationary.